component/org.wso2.carbon.identity.verification.evident.connector/src/main/java/org/wso2/carbon/identity/verification/evident/EvidentIDVHandler.java

```diff
@@ -0,0 +1,541 @@
+        String username = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+        UserStoreManager userStoreManager =
+                (UserStoreManager) eventProperties.get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+
+        String tenantDomain = (String) eventProperties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+
+        Property[] identityProperties;
+        try {
+            identityProperties = EvidentIDVDataHolder.getInstance()
+                    .getIdentityGovernanceService().getConfiguration(getPropertyNames(), tenantDomain);
+        } catch (IdentityGovernanceException e) {
+            throw new IdentityEventException("Error while retrieving account lock handler properties.", e);
+        }
+
+        boolean enabled = false;
+
+        for (Property property : identityProperties) {
+            if (EvidentIDVConstants.EVIDENT_VERIFICATION_ENABLE.equals(property.getName())) {
+                enabled = Boolean.parseBoolean(property.getValue());
+            }
+            if (EvidentIDVConstants.EVIDENT_API_KEY.equals(property.getName())) {
+                key = property.getValue();
+            }
+            if (EvidentIDVConstants.EVIDENT_API_SECRET.equals(property.getName())) {
+                secret = property.getValue();
+            }
+            if (EvidentIDVConstants.EVIDENT_API_BASE_PATH.equals(property.getName())) {
+                basePath = property.getValue();
+            }
+            if (EvidentIDVConstants.EVIDENT_USER_STORES.equals(property.getName())) {
+                userStores = property.getValue();
+            }
+            if (EvidentIDVConstants.EVIDENT_EMAIL_SUMMARY.equals(property.getName())) {
+                emailSummary = property.getValue();
+            }
+            if (EvidentIDVConstants.EVIDENT_EMAIL_DESCRIPTION.equals(property.getName())) {
+                emailDescription = property.getValue();
+            }
+        }
+
+        if (enabled) {
+            // Property validation
+            if (StringUtils.isEmpty(key) || StringUtils.isEmpty(secret) || StringUtils.isEmpty(basePath)) {
+                log.warn("Evident identity verification is enabled but one or more required parameters are not " +
+                        "provided");
+                return;
+            }
+
+            // Validate user store
+            if (StringUtils.isNotEmpty(userStores)) {
+                List userStoreList = Arrays.asList(userStores.trim().split("\\s*,\\s*"));
+                String currentUserStore = userStoreManager.getRealmConfiguration().getUserStoreProperties().get(
+                        UserStoreConfigConstants.DOMAIN_NAME);
+                if (!userStoreList.contains(currentUserStore)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("User: " + username + " is trying to login. Returning since the user store: " +
```

Comment:
Is this always, login event? since it could be posted add user as well can we change the log accordingly?